core/test/unit/services/themes/middleware_spec.js

```diff
@@ -1,208 +1,154 @@
-                    templateOptions.data.config.should.be.an.Object();
-                    // posts per page should NOT be set as there's no active theme
-                    should.not.exist(templateOptions.data.config.posts_per_page);
+    it('calls updateTemplateOptions with correct data', function (done) {
+        const themeDataExpectedProps = ['posts_per_page', 'image_sizes'];
 
-                    // Check blog config tried to call public settings
-                    settingPublicStub.calledOnce.should.be.true();
+        executeMiddleware(middleware, req, res, function next(err) {
+            should.not.exist(err);
 
-                    // url should be correct
-                    templateOptions.data.blog.url.should.eql('http://127.0.0.1:2369');
+            hbs.updateTemplateOptions.calledOnce.should.be.true();
+            const templateOptions = hbs.updateTemplateOptions.firstCall.args[0];
+            const data = templateOptions.data;
 
-                    // Check labs config
-                    templateOptions.data.labs.should.be.an.Object();
+            data.should.be.an.Object().with.properties('site', 'blog', 'labs', 'config');
 
-                    done();
-                });
-            });
+            // Check Theme Config
+            data.config.should.be.an.Object()
+                .with.properties(themeDataExpectedProps)
+                .and.size(themeDataExpectedProps.length);
+            // posts per page should be set according to the stub
+            data.config.posts_per_page.should.eql(2);
 
-            it('calls updateTempalateOptions with correct info for secure context', function (done) {
-                req.secure = true;
+            // Check labs config
+            should.deepEqual(data.labs, fakeLabsData);
 
-                updateTemplateData(req, res, function next(err) {
-                    var templateOptions;
-                    should.not.exist(err);
+            should.equal(data.site, fakeSiteData);
+            should.equal(data.blog, fakeSiteData);
 
-                    updateOptionsStub.calledOnce.should.be.true();
-                    templateOptions = updateOptionsStub.firstCall.args[0];
-                    templateOptions.should.be.an.Object().with.property('data');
-                    templateOptions.data.should.be.an.Object().with.properties('blog', 'labs', 'config');
+            done();
+        });
+    });
 
-                    // Check Theme Config
-                    templateOptions.data.config.should.be.an.Object()
-                        .with.properties(themeDataExpectedProps)
-                        .and.size(themeDataExpectedProps.length);
-                    // posts per page should be set according to the stub
-                    templateOptions.data.config.posts_per_page.should.eql(2);
+    it('Sets res.locals.secure to the value of req.secure', function (done) {
+        req.secure = Math.random() < 0.5;
```

Comment:
Interesting approach :+1:

Reply:
:grin: Probably not the best, but for a boolean it's kinda fun to have :man_shrugging: